numpy/core/tests/test_records.py

```diff
@@ -332,7 +332,8 @@ class TestPathUsage(object):
     def test_tofile_fromfile(self):
         with temppath(suffix='.bin') as path:
             path = Path(path)
-            a = np.empty(10, dtype='f8,i4,a5')
+            np.random.seed(123)
+            a = np.random.rand(10).astype('f8,i4,a5')
             a[5] = (0.5,10,'abcde')
             a.newbyteorder('<')
```

Comment:
AFAICT, this doesn't change anything about `a`.

Reply:
It returns a new array. I think it can be omitted and the read should default to native order. Reversing the order of the original could also lead to problematic floats.

Reply:
I've revised to remove the byte ordering stuff.

The purpose of setting `a[5]` might be to put some more "interesting" data into the array for the purpose of the roundtrip--if empty or random put a bunch of zeros in there I suppose one might argue it is a slightly less thorough check than i.e., heterogeneous data. Indeed, with my change, the integer field becomes all zero before that I think, because of the casting.

But that's just speculation -- I didn't write the original test of course.

Reply:
> I didn't write the original test of course.

Yeah, the original tests seems more a test of records than path :)